src/Microsoft.AspNetCore.Hosting/Startup/StartupBase.cs

```diff
@@ -16,4 +16,25 @@ public virtual IServiceProvider ConfigureServices(IServiceCollection services)
             return services.BuildServiceProvider();
         }
     }
+
+    public abstract class StartupBase<TContainer> : IStartup
+    {
+        private readonly IServiceProviderFactory _factory;
+
+        public StartupBase(IServiceProviderFactory factory)
+        {
+            _factory = factory;
+        }
+
+        public abstract void Configure(IApplicationBuilder app);
+
+        public IServiceProvider ConfigureServices(IServiceCollection services)
```

Comment:
Why not make it explicit and have a `void ConfigureServices(IServiceCollection services)`, like [this](https://github.com/aspnet/Hosting/commit/7556d29381df67a69ee84b688a6a05b2e8825518#diff-4c7c346129759b4849377b3494d5b17fR40)?